wagtail/images/models.py

```diff
@@ -319,93 +328,166 @@ def get_rendition_model(cls):
         """Get the Rendition model for this Image model"""
         return cls.renditions.rel.related_model
 
-    def get_rendition(self, filter):
+    def get_rendition(self, filter: Union["Filter", str]) -> "AbstractRendition":
+        """
+        Returns a ``Rendition``* instance with a ``file`` field value (an
+        image) reflecting the supplied ``filter`` value and focal point values
+        from this object.
+
+        *If using custom image models, an instance of the custom rendition
+        model will be returned.
+        """
         if isinstance(filter, str):
             filter = Filter(spec=filter)
 
-        cache_key = filter.get_cache_key(self)
         Rendition = self.get_rendition_model()
 
         try:
-            rendition_caching = True
+            rendition = self.find_existing_rendition(filter)
+        except Rendition.DoesNotExist:
+            rendition = self.create_rendition(filter)
+            # Reuse this rendition if requested again from this object
+            if "renditions" in getattr(self, "_prefetched_objects_cache", {}):
+                self._prefetched_objects_cache["renditions"]._result_cache.append(
```

Comment:
This feels a little strange to me. Normally newly created models wouldn't be inserted magically into the prefetch cache. Is there a use case this is trying to cover?

Would also be great to see a test case covering this usage.

Reply:
@RealOrangeOne Already got you covered: https://github.com/wagtail/wagtail/pull/8069/files#diff-4d34f3d99a715020023655d4191cf700a6b9d2cbefd0c2a0aff4e8a16f098474R306-R311

Reply:
As for a use-case: Imagine you have awkward design requirements, where the list of things needs to be rendered twice in different parts of the page, so that things appear in a more desirable order on mobile VS desktop. Or maybe there could be similar requirements for the individually rendered items. In both of these cases, this should prevent the same image rendition being freshly generated multiple times.